packages/lib/isOutOfBounds.tsx

```diff
@@ -59,121 +63,142 @@ export function calculatePeriodLimits({
       const rollingEndDay = periodCountCalendarDays
-        ? currentTime.add(periodDays, "days")
-        : currentTime.businessDaysAdd(periodDays);
+        ? currentTimeInBookerTz.add(periodDays, "days")
+        : currentTimeInBookerTz.businessDaysAdd(periodDays);
       // The future limit talks in terms of days so we take the end of the day here to consider the entire day
-      return { rollingEndDay: rollingEndDay.endOf("day"), rangeStartDay: null, rangeEndDay: null };
+      return {
+        endOfRollingPeriodEndDayInBookerTz: rollingEndDay.endOf("day"),
+        startOfRangeStartDayInEventTz: null,
+        endOfRangeEndDayInEventTz: null,
+      };
     }
 
     case PeriodType.ROLLING_WINDOW: {
       if (_skipRollingWindowCheck) {
-        return { rollingEndDay: null, rangeStartDay: null, rangeEndDay: null };
+        return {
+          endOfRollingPeriodEndDayInBookerTz: null,
+          startOfRangeStartDayInEventTz: null,
+          endOfRangeEndDayInEventTz: null,
+        };
       }
 
-      if (!allDatesWithBookabilityStatus) {
+      if (!allDatesWithBookabilityStatusInBookerTz) {
         throw new Error("`allDatesWithBookabilityStatus` is required");
       }
 
-      const rollingEndDay = getRollingWindowEndDate({
-        startDate: currentTime,
+      const endOfRollingPeriodEndDayInBookerTz = getRollingWindowEndDate({
+        startDateInBookerTz: currentTimeInBookerTz,
         daysNeeded: periodDays,
-        allDatesWithBookabilityStatus,
+        allDatesWithBookabilityStatusInBookerTz,
         countNonBusinessDays: periodCountCalendarDays,
       });
 
-      return { rollingEndDay, rangeStartDay: null, rangeEndDay: null };
+      return {
+        endOfRollingPeriodEndDayInBookerTz,
+        startOfRangeStartDayInEventTz: null,
+        endOfRangeEndDayInEventTz: null,
+      };
     }
 
     case PeriodType.RANGE: {
-      // The future limit talks in terms of days so we take the start of the day for starting range and endOf the day for ending range
-      const rangeStartDay = dayjs(periodStartDate).utcOffset(utcOffset).startOf("day");
-      const rangeEndDay = dayjs(periodEndDate).utcOffset(utcOffset).endOf("day");
+      // We take the start of the day for the start of the range and endOf the day for end of range, so that entire days are covered
+      // We use organizer's timezone here(in contrast with ROLLING/ROLLING_WINDOW where number of days is available and not the specific date objects).
+      // This is because in case of range the start and end date objects are determined by the organizer, so we should consider the range in organizer/event's timezone.
+      const startOfRangeStartDayInEventTz = dayjs(periodStartDate).utcOffset(eventUtcOffset).startOf("day");
+      const endOfRangeEndDayInEventTz = dayjs(periodEndDate).utcOffset(eventUtcOffset).endOf("day");
```

Comment:
This is what changed. It is using event timezone instead of booker timezone now.